notifications_api_common/tasks.py

```diff
@@ -0,0 +1,61 @@
+import logging
+
+import celery
+import requests
+from zds_client import ClientError
+
+from .models import NotificationsConfig
+
+logger = logging.getLogger(__name__)
+
+
+class ConfigTask(celery.Task):
+    """
+    add retry options from NotificationsConfig into the task instance
+    """
+
+    def __init__(self):
+        config = NotificationsConfig.get_solo()
+
+        self.max_retries = config.notification_delivery_max_retries
+        self.retry_backoff = config.notification_delivery_retry_backoff
+        self.retry_backoff_max = config.notification_delivery_retry_backoff_max
```

Comment:
The docs say that a task is only instantiated once per process (see https://docs.celeryq.dev/en/stable/userguide/tasks.html#instantiation). Effectively, that means that when you change these values in the admin, that you need to restart the celery workers for the changes to take effect.

Looking at the example given in https://docs.celeryq.dev/en/stable/userguide/tasks.html#automatic-retry-for-known-exceptions, maybe a better approach is using a descriptor.

Example:

```py
class BaseTaskWithRetry(Task):
    autoretry_for = (TypeError,)
    max_retries = 5
    retry_backoff = True
    retry_backoff_max = 700
    retry_jitter = False
```

Our variant:

```py
class from_config:
    def __set_name__(self, owner, name: str):
        self.config_field = f"notification_delivery_{name}"

    def __get__(self, obj, objtype=None):
        config = NotificationsConfig.get_solo()
        return getattr(config, self.config_field)


class ConfigTask(Task):
    max_retries = from_config()
    retry_backoff = from_config()
    retry_backoff_max = from_config()
```

Reply:
discussed with Sergei and decided to restore Steven's implementation, because 
patching `Task.run` method seems the only solution to change autoretry properties during runtime